sympy/solvers/solvers.py

```diff
@@ -2674,7 +2674,13 @@ def _tsolve(eq, sym, **flags):
                     return _solve(lhs.exp, sym, **flags) if rhs != 0 else []
 
                 # Gets most solutions...
-                sol = _solve(exp(log(lhs.base)*lhs.exp)-exp(log(rhs)), sym, **flags)
+                if lhs.base == rhs.as_base_exp()[0]:
```

Comment:
You are allowing for rhs to not be a power when you do it like this...and if it's not a power the next line will fail for lack of exp attribute. Do you need to differentiate between `b**f(x) - b` and `b**f(x) - b**a`?

Reply:
Sorry i forgot to change `rhs.exp` to `rhs.as_base_exp()[1]`. BTW I couldn't understand your question.